### PR TITLE
chore: Improve testing

### PR DESCRIPTION
**Pull request by dnys1**
Use `build_test` package to test both DDC and dart2js in CI which will help catch discrepancies earlier in our dev cycle. This requires separating out the tests into a new, private package so that build_runner can correctly discern the dependency graph.

### CODE REVIEW

File: packages/auth/amplify_auth_cognito_test/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+Tests for the [`amplify_auth_cognito_dart`](../amplify_auth_cognito_dart) package separated out so that they can be easily tested using `build_runner`.
+
+## Testing
+
+To run tests:
+
+```sh
+$ dart test
+$ dart run build_runner test --delete-conflicting-outputs -- -p chrome,firefox
+$ dart run build_runner test --release --delete-conflicting-outputs -- -p chrome,firefox
```
Comment (haverchuck) [inline packages/auth/amplify_auth_cognito_test/README.md:10]: Can this work with Edge, like Flutter tests? If so, might be worth noting somewhere.

**Review by haverchuck — APPROVED** [review]
Left one nit, but otherwise looks great.

File: packages/auth/amplify_auth_cognito_test/README.md (hunk 1 above)
Comment (dnys1) [reply packages/auth/amplify_auth_cognito_test/README.md:10]: Oh yeah, I forget about edge 😂 I'll see if we can get it running on windows

Comment (dnys1) [reply packages/auth/amplify_auth_cognito_test/README.md:10]: Looks like it's not supported yet, unfortunately: https://github.com/dart-lang/test/issues/1142. The Flutter team jumped through some hoops to get it to work (https://github.com/flutter/engine/pull/15641) and I'm not sure if that effort is worth it right now.

Comment (haverchuck) [reply packages/auth/amplify_auth_cognito_test/README.md:10]: Ok, sounds good

**Review by Jordan-Nelson — APPROVED** [review]

File: packages/secure_storage/amplify_secure_storage_test/mono_pkg.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  - unit_test:
+      - command: dart --version
+      - command: tool/test-desktop.sh
+        os:
+          - windows
+          - macos
+          - linux
+      # TODO(dnys1): Secure storage will not work in DDC right now due to ongoing staticInterop changes. 
+      # Since the interfaces are only ever called via workers (dart2js), this is not a problem but should be monitored.
+      # Issue: https://github.com/dart-lang/sdk/issues/49301
```
[the remark this answers is not in the document]
Comment (dnys1) [reply packages/secure_storage/amplify_secure_storage_test/mono_pkg.yaml:19]: I'll double check - may only be fixed in dev

**Pull request merged**